torch_xla/csrc/xla_sharding_util.cpp

```diff
@@ -382,4 +383,60 @@ std::vector<at::Tensor> ShardingUtil::ShardTensor(
   return shards;
 }
 
+void ShardingUtil::PrepareOutputShardingPropagation(
+    std::vector<XLATensorPtr>* tensors, absl::Span<const size_t> indices,
+    ComputationPtr computation,
+    std::vector<torch::lazy::BackendDataPtr>* data_placeholders,
+    std::vector<XLATensor::ShardingSpecPtr>* sharding_specs) {
+  auto computation_proto = computation->computation().proto();
+  auto tuple_shardings =
+      computation_proto.spmd_output_sharding().tuple_shardings();
+  if (tuple_shardings.empty()) {
+    return;
+  }
+  XLA_CHECK(indices.size() == tuple_shardings.size())
+      << "Expected " << indices.size() << " sharding annotations, but got "
+      << tuple_shardings.size();
+
+  for (int i = 0; i < indices.size(); ++i) {
+    auto xtensor = (*tensors)[indices[i]];
+    if (tuple_shardings[i].type()) {
```

Comment:
Could you add a comment explaining what this is testing?

Reply:
```
// Output parameter sharding annotations, defaults to REPLICATED(0) if unset.
auto tuple_shardings = ...
...
if (tuple_shardings[i].type()) {
      // Tensor sharding annotation type is non-zero (sharded).
```